### PR TITLE
Refactor all API routes

### PR DESCRIPTION
**Pull request by hnguyen0**
restructured backend routes and made api calls in the frontend

~should~~ be updated with [API Doc](https://docs.google.com/document/d/1HvOvkkIZIm1OWNSsGH8ve0ib5ygfrsqwzop11SNMFFc/edit#)

API's to fill out:
- "/download" in csv.js
- "retrieve/all" in period.js
- signout?

Note Frontend: all the backend api's are now connected and called in the frontend api folder so just need to call those functions

### CODE REVIEW

File: api/routes/behavior.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ router.get("/count-list/:student_id", function (req, res) {
         const options = {
             projection: {
                 _id: 0,
-                date: 0,
+                time: 0,
```
Comment (Drakran) [inline api/routes/behavior.js:53]: sort of sus

**Review by Drakran — APPROVED** [review]
sort of sus but i approve of pretty code but still sus

File: client/src/api/behavior.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export async function addBehavior(studentId, behaviorName) {
+    const res = await fetch("behavior/add/" + studentId, {
+        method: "GET",
```
Comment (justinyh) [inline client/src/api/behavior.js:3]: if we're "adding a behavior" shouldn't the method be PUT or POST (probably POST?)?

Comment (hnguyen0) [reply client/src/api/behavior.js]: Oh good catch! Backend is POST so I fixed it to POST ~~Thanks!

**Review by justinyh — APPROVED** [review]
Looks good to me as long as everything works!

**Pull request merged**